platforms/android/tangram/src/main/java/com/mapzen/tangram/MapController.java

```diff
@@ -1089,24 +1088,66 @@ public void removeAllMarkers() {
      */
     public void setMapChangeListener(@Nullable final MapChangeListener listener) {
         mapChangeListener = listener;
+        mapRegionChangeStateManager = new MapRegionChangeStateManager(mapChangeListener);
     }
 
-    //Convenience member functions
-    private void onRegionWillChange(boolean animated) {
-        if (mapChangeListener != null) {
-            mapChangeListener.onRegionWillChange(animated);
+    private static class MapRegionChangeStateManager {
+
+        enum MapRegionChangeState {
+            MAP_REGION_NO_CHANGE,
+            MAP_REGION_WILL_CHANGE,
```

Comment:
Do you think it would be easier to think of change events as transitions rather than states? The states you have here seem to me like transitions between "idle", "animating", and "jumping".

Reply:
Hmm, it might be actually, will give it a try.

Reply:
Gave this some more thought, I think it is simpler and more logical to use transitions instead of the `idle`, `animating` and `jumping` states.
Transitions seems to match exactly with the map events we are catering to here, also can help reduce some logic, example:
- One should not repeated call `mapChangeListener.onRegionWillChange(animated)` is the transition is already `WILL_CHANGE_ANIMTED` or `WILL_CHANGE`. I do not think this can be aptly mapped with the states.

Reply:
I can however change the enum name to `MapRegionChangeTransitions`?

Reply:
It was easy enough to implement on top of your changes so I wrote out the logic as I've been thinking of it: https://github.com/tangrams/tangram-es/commit/15e1859cb1ddbef181b8cf8b5275bbb9a9eb7d92

Reply:
Hmm I see, will see if I can extend that logic. Example if I start panning  the map when map was animating say due to a flyTo, or if some event explicitly sets a position (basically state set to jumping) when map is animating. Such state management logic will have to be added to this.

Reply:
Not claiming that this is objectively better in any way, I just wanted to see how it would work.

Reply:
Ohh totally, I actually like lesser enums (states vs transitions) getting us to the same place. So will be a cleaner approach.

Reply:
@matteblair adopted States here: 53fb10f3ed0e1d64a4a1d33a617556c9a5e3f000